src/vmm/src/builder.rs

```diff
@@ -553,9 +549,9 @@ pub fn setup_interrupt_controller(
 /// Sets up the serial device.
 pub fn setup_serial_device(
     event_manager: &mut EventManager,
-    input: Box<dyn devices::legacy::ReadableFd + Send>,
     out: Box<dyn io::Write + Send>,
 ) -> super::Result<Arc<Mutex<Serial>>> {
+    let input = Box::new(SerialStdin::get());
```

Comment:
I'm unsure about this because it removes an otherwise generic and configurable input source (`ReadableFd`) and replaces it with the very specific `SerialStdin`, which can't be substituted by anything else. I'm thinking of a user that consumes the programmatic API (which this fn is a part of, and which should stay flexible) and wants to plugin a fancy custom serial console that's not our `SerialStdin`.

Reply:
How about making `SerialStdin` `pub(crate)`?

Reply:
Yup, I made it `pub (crate)`, take a look.